src/main/java/net/sf/jabref/BibDatabaseContext.java

```diff
@@ -175,13 +175,14 @@ private String getFileDirectoryPath(String directoryName) {
         String dir = directoryName;
         // If this directory is relative, we try to interpret it as relative to
         // the file path of this BIB file:
-        if (!new File(dir).isAbsolute() && (getDatabaseFile() != null)) {
+        Optional<File> databaseFile = getDatabaseFile();
+        if (!new File(dir).isAbsolute() && databaseFile.isPresent()) {
```

Comment:
OK. I have very little (basically no) idea about nio.Paths though, so I prefer to avoid it at this time... If you could spare some time and provide code I can just paste it would be appreciated though.